flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobVertexBackPressureInfo.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.rest.messages;
+
+import org.apache.flink.runtime.rest.handler.job.JobVertexBackPressureHandler;
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonCreator;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonInclude;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonValue;
+
+import java.util.List;
+import java.util.Objects;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Response type of the {@link JobVertexBackPressureHandler}.
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class JobVertexBackPressureInfo implements ResponseBody {
+
+	public static final String FIELD_NAME_STATUS = "status";
+	public static final String FIELD_NAME_BACKPRESSURE_LEVEL = "backpressure-level";
+	public static final String FIELD_NAME_END_TIMESTAMP = "end-timestamp";
+	public static final String FIELD_NAME_SUBTASKS = "subtasks";
+
+	@JsonProperty(FIELD_NAME_STATUS)
+	private final VertexBackPressureStatus status;
+
+	@JsonProperty(FIELD_NAME_BACKPRESSURE_LEVEL)
+	private final VertexBackPressureLevel backpressureLevel;
+
+	@JsonProperty(FIELD_NAME_END_TIMESTAMP)
+	private final Long endTimestamp;
+
+	@JsonProperty(FIELD_NAME_SUBTASKS)
+	protected final List<SubtaskBackPressureInfo> subtasks;
+
+	@JsonCreator
+	public JobVertexBackPressureInfo(
+		@JsonProperty(FIELD_NAME_STATUS) VertexBackPressureStatus status,
+		@JsonProperty(FIELD_NAME_BACKPRESSURE_LEVEL) VertexBackPressureLevel backpressureLevel,
+		@JsonProperty(FIELD_NAME_END_TIMESTAMP) Long endTimestamp,
+		@JsonProperty(FIELD_NAME_SUBTASKS) List<SubtaskBackPressureInfo> subtasks) {
+		this.status = status;
+		this.backpressureLevel = backpressureLevel;
+		this.endTimestamp = endTimestamp;
+		this.subtasks = subtasks;
+	}
+
+	public static JobVertexBackPressureInfo deprecated() {
+		return new JobVertexBackPressureInfo(
+			VertexBackPressureStatus.DEPRECATED,
```

Comment:
Why do we explicitly say that the back pressure information is deprecated. Serving some old information is better than nothing if we still have some older back pressure information, right? The caller could still throw the information away based on `endTimestamp` if it is too old.

Reply:
Can do and I thought about it but that's how it is done currently. If status is `deprecated`, the ui will render *Sampling in progress*.

Reply:
done